source/client/process_impl.cc

```diff
@@ -192,7 +198,8 @@ void ProcessImpl::configureComponentLogLevels(spdlog::level::level_enum level) {
 }
 
 uint32_t ProcessImpl::determineConcurrency() const {
-  uint32_t cpu_cores_with_affinity = Envoy::OptionsImplPlatform::getCpuCount();
+  // Spare 1 cpu for flush worker.
+  uint32_t cpu_cores_with_affinity = Envoy::OptionsImplPlatform::getCpuCount() - 1;
```

Comment:
@oschaaf Do we need to reserve a cpu core for the flush worker?

Reply:
I think this is a very good question. IMHO, the main thread and the flush worker would probably best be combined on a distinct dedicated core (if that luxury is an option): 
- this makes sure that the workers responsible for generating load won't be disturbed
- cpu consumption will be easily observable as spikes won't jump from core to core. chances of problematic behaviour slipping by unnoticed will be drastically lower.
Having said that, we don't do anything special here for the main thread either (but that's pretty much idle during load generation). Maybe we should discuss this on Slack for a bit with other devs to hear what they think about this?

Reply:
I think that maybe the right thing to do with respect to the point you raised, is maybe punt this to an issue, and open a discussion on Slack or some such?

Reply:
Thanks Otto. Opened https://github.com/envoyproxy/nighthawk/issues/441
Also reverted the change for now.